app/main.py

```diff
@@ -1,34 +1,70 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = True
+
+    def hit(self) -> None:
+        self.is_alive = False
+
+    def __str__(self) -> str:
+        return "*" if not self.is_alive else u"\u25A1"
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple[int, int], end: tuple[int, int]) -> None:
+        self.start = start
+        self.end = end
+        self.decks = self._create_decks(start, end)
+        self.is_drowned = False
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    @staticmethod
+    def _create_decks(start: tuple[int, int],
+                      end: tuple[int, int]) -> list[Deck]:
+        decks = []
+        if start[0] == end[0]:
+            for col in range(start[1], end[1] + 1):
+                decks.append(Deck(start[0], col))
+        else:
+            for row in range(start[0], end[0] + 1):
+                decks.append(Deck(row, start[1]))
+        return decks
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> str:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                deck.hit()
+                self.is_drowned = all(not d.is_alive for d in self.decks)
+                return "Sunk!" if self.is_drowned else "Hit!"
+        return "Miss!"
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list[Ship]) -> None:
+        self.ships = [Ship(start, end) for start, end in ships]
+        self.field = self._create_field()
+        self._validate_field()
+
+    def _create_field(self) -> list[list[str]]:
+        field = [["~"] * 10 for _ in range(10)]
```

Comment:
The field size is hardcoded to 10x10. If the game rules allow for different field sizes, consider parameterizing this value or documenting the constraint.